### PR TITLE
[cycraft-2090] CyberTotal

### PR DESCRIPTION
**Pull request by bpyliu**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This is integration of CyberTotal and Demisto, includes 7 commands and 2 playbooks. 

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [x] The title must be in the following format: **[cycraft-2090] This integration serves to enrich XSOAR indicators including IP addresses, URLs, domains, and file hashes with CyberTotal information concerning reputation and whois data for the purpose of triaging, validating, deciding action on and otherwise enriching indicators in XSOAR. **

### CODE REVIEW

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 1]
```diff
@@ -0,0 +1,845 @@
...
+      description: The name of registrar
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.registrarEmail
+      description: The email address of registrar
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.registrarPhone
+      description: The phone number of registrar
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.registrarCreatedAt
+      description: 'Registrar create date format: YYYY-MM-DD HH:mm:SS '
```
Comment (hod-alpert) [inline Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:435]: This comes back as epoch time according to the test_data
Either parse it to ISO format, or change documentation
```suggestion
      description: Registrar create date format: ISO 8601
```

**Comment by hod-alpert** [comment]
When you want to parse a UNIX timestamp called for example `timestamp`
simply do:
```
datetime.fromtimestamp(timestamp, timezone.utc).isoformat()
```

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+            return_results(ip_whois_command(client, demisto.args()))
+
+        elif demisto.command() == 'cybertotal-url-whois':
+            return_results(url_whois_command(client, demisto.args()))
+
+        elif demisto.command() == 'cybertotal-domain-whois':
+            return_results(domain_whois_command(client, demisto.args()))
+
+        # elif demisto.command() == 'whois':
+        #     return_results(whois_command(client, demisto.args(), wait_inprogress_interval))
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py:826]: It's just unused code. I'll remove it.

File: Packs/CyberTotal/Integrations/CyberTotal/README.md [hunk 3]
```diff
@@ -0,0 +1,991 @@
+CyberTotal is a cloud-based threat intelligence service developed by CyCraft.
+This integration was integrated and tested with version xx of CyberTotal
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/README.md:2]: I'll fill in a version.

File: Packs/CyberTotal/TestPlaybooks/CyberTotal_TestPlaybook.yml [hunk 4]
```diff
@@ -0,0 +1,851 @@
...
+      iscommand: true
+      brand: CyberTotal
+    nexttasks:
+      '#none#':
+      - "3"
+    scriptarguments:
+      ip:
+        complex:
+          root: IP
+          accessor: Address
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/TestPlaybooks/CyberTotal_TestPlaybook.yml:88]: I'll update the test playbook.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py [hunk 5]
```diff
@@ -0,0 +1,838 @@
...
+                        positive_detections = positive_detections + 1
+        result['detection_engines'] = detection_engines
+        result['positive_detections'] = positive_detections
+        result['detection_ratio'] = str(positive_detections) + '/' + str(detection_engines)
+        result['message'] = 'search success'
+        if 'score' in cybertotal_result['basic']:
+            result['severity'] = cybertotal_result['basic']['score'].get('severity', -1)
+            result['confidence'] = cybertotal_result['basic']['score'].get('confidence', -1)
+            result['threat'] = cybertotal_result['basic']['score'].get('threat', '')
+        return result
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py [hunk 6]
```diff
@@ -0,0 +1,838 @@
...
+        if 'whois' in cybertotal_result:
+            if len(cybertotal_result['whois']) > 0:
+                result = cybertotal_result['whois'].pop(0)
+
+        result['permalink'] = permalink,
+        result['resource'] = url,
+        result['scan_date'] = dateparser.parse(scan_time).strftime("%Y-%m-%d %H:%M:%S"),
+        result['task_id'] = task_id
+        result["message"] = "search success"
+        return result
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py:306]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml (hunk 1 above)
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:435]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 7]
```diff
@@ -0,0 +1,845 @@
...
+      description: Message about this search.
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.permalink
+      description: The link of this whois report in CyberTotal.
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.createdAt
+      description: 'Create date format: YYYY-MM-DD HH:mm:SS '
+      type: date
+    - contextPath: CyberTotal.WHOIS-IP.updatedAt
+      description: 'Update date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:408]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 8]
```diff
@@ -0,0 +1,845 @@
...
+  - name: cybertotal-ip-whois
+    arguments:
+    - name: ip
+      required: true
+      default: true
+      description: List of IP(s).
+      isArray: true
+    outputs:
+    - contextPath: CyberTotal.WHOIS-IP.scan_date
+      description: 'Scan date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:390]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 9]
```diff
@@ -0,0 +1,845 @@
...
+      description: List of url(s).
+      isArray: true
+    - name: threshold
+      description: If the URL has reputation above the threshold then the URL defined
+        as malicious. If threshold not set, then threshold from instance configuration
+        is used.
+      defaultValue: "10"
+    outputs:
+    - contextPath: CyberTotal.URL.scan_date
+      description: 'Scan date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:319]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 10]
```diff
@@ -0,0 +1,845 @@
...
+      description: List of domains.
+      isArray: true
+    - name: threshold
+      description: If the domain has reputation above the threshold then the domain
+        defined as malicious. If threshold not set, then threshold from instance configuration
+        is used.
+      defaultValue: "10"
+    outputs:
+    - contextPath: CyberTotal.Domain.scan_date
+      description: 'Scan date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:246]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 11]
```diff
@@ -0,0 +1,845 @@
...
+      description: list of hash(s).
+      isArray: true
+    - name: threshold
+      description: If the HASH has reputation above the threshold then the HASH defined
+        as malicious. If threshold not set, then threshold from instance configuration
+        is used.
+      defaultValue: "10"
+    outputs:
+    - contextPath: CyberTotal.File.scan_date
+      description: 'Scan date format: YYYY-MM-DD HH:mm:SS'
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:150]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 12]
```diff
@@ -0,0 +1,845 @@
...
+      description: The scan target sent to CyberTotal.
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.message
+      description: Message about this search.
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.permalink
+      description: The link of this whois report in CyberTotal.
+      type: string
+    - contextPath: CyberTotal.WHOIS-IP.createdAt
+      description: 'Create date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:405]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml [hunk 13]
```diff
@@ -0,0 +1,845 @@
...
+      description: List of IPs.
+      isArray: true
+    - name: threshold
+      description: If the IP has reputation above the threshold then the IP defined
+        as malicious. If threshold not set, then threshold from instance configuration
+        is used.
+      defaultValue: "10"
+    outputs:
+    - contextPath: CyberTotal.IP.scan_date
+      description: 'Scan date format: YYYY-MM-DD HH:mm:SS '
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.yml:74]: I'll improve it.

File: Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py [hunk 14]
```diff
@@ -0,0 +1,838 @@
...
+        client = Client(
+            base_url=cybertotal_url,
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy)
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            # result = test_module(client, first_fetch_time)
+            # return_results(result)
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/CyberTotal.py:797]: I'll improve it.

**Comment by hod-alpert** [comment]
@michalgold we have two playbooks here, can someone from your team check them?

File: Packs/CyberTotal/Integrations/CyberTotal/README.md [hunk 15]
```diff
@@ -0,0 +1,997 @@
...
...DNS Hosting.\n\nNOTICE: You are not authorized to access or query our WHOIS database through the use of high-volume, automated, electronic processes or for the purpose or purposes of using the data in any manner that violates these terms of use. The Data in the CSC WHOIS database is provided by CSC for information purposes only, and to assist persons in obtaining information about or related to a domain name registration record. CSC does not guarantee its accuracy. By submitting a WHOIS query, you agree to abide by the following terms of use: you agree that you may use this Data only for lawful purposes and that under no circumstances will you use this Data to: (1) allow, enable, or otherwise support the transmission of mass unsolicited, commercial advertising or solicitations via direct mail, e-mail, telephone, or facsimile; or (2) enable high volume, automated, electronic processes that apply to CSC (or its computer systems). CSC reserves the right to terminate your access to the WHOIS database in its sole discretion for any violations by you of these terms of use. CSC reserves the right to modify these terms at any time.\n\nRegister your domain name at http://www.cscglobal.com",
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/README.md:718]: ok we will remove the raw response.

File: Packs/CyberTotal/Integrations/CyberTotal/README.md [hunk 16]
```diff
@@ -0,0 +1,997 @@
...
```
[the remark this answers is not in the document]
Comment (bpyliu) [reply Packs/CyberTotal/Integrations/CyberTotal/README.md:931]: as the previous

**Comment by hod-alpert** [comment]
@bpyliu Once you'll deliver the PR fixes, I'll merge it

**Pull request merged**